plugin-server/src/utils/db/db.ts

```diff
@@ -878,19 +878,29 @@ export class DB {
         return client ? kafkaMessages : updatedPerson
     }
 
-    public async deletePerson(person: Person, client: PoolClient): Promise<ProducerRecord[]> {
-        await client.query('DELETE FROM posthog_person WHERE team_id = $1 AND id = $2', [person.team_id, person.id])
-        const kafkaMessages = [
-            generateKafkaPersonUpdateMessage(
-                person.created_at,
-                person.properties,
-                person.team_id,
-                person.is_identified,
-                person.uuid,
-                null,
-                1
-            ),
-        ]
+    public async deletePerson(person: Person, client?: PoolClient): Promise<ProducerRecord[]> {
+        const result = await this.postgresQuery<{ version: string }>(
+            'DELETE FROM posthog_person WHERE team_id = $1 AND id = $2 RETURNING version',
+            [person.team_id, person.id],
+            'deletePerson',
+            client
+        )
+
+        let kafkaMessages: ProducerRecord[] = []
+
+        if (result.rows.length > 0) {
+            kafkaMessages = [
+                generateKafkaPersonUpdateMessage(
+                    person.created_at,
+                    person.properties,
+                    person.team_id,
+                    person.is_identified,
+                    person.uuid,
+                    Number(result.rows[0].version || 0) + 1,
```

Comment:
didn't we have a discussion before about how this should be the max possible value?

Reply:
that could be problematic as the update can still happen afterwards & what would we result with then, but for extra safety it might be good to do `+1000`

Reply:
Added +100

Reply:
?

I assume once a person has been deleted they don't get brought back, so we should probably ensure that as strongly as we can.

Reply:
I think @tiina303 found a different bug here [thread](https://github.com/PostHog/posthog/pull/10207#discussion_r893311080) - let's document this and fix separately. 

Also pushed which I forgot to earlier :sweat_smile:

Reply:
yup, still adding +100 like karl did here adds a bit of extra safety net for us 👍  I'll put up a fix for the other problem I found.